src/main/duke/command/AddCommand.java

```diff
@@ -0,0 +1,58 @@
+package duke.command;
+
+import duke.TaskList;
+import duke.Ui;
+import duke.Storage;
+import duke.DukeException;
+import duke.task.Task;
+import duke.task.TodoTask;
+import duke.task.DeadlineTask;
+import duke.task.EventTask;
+
+public class AddCommand extends Command {
+
+    private Task taskToAdd;
+    
+    public AddCommand(String fullCommand) throws DukeException{
+        parseTask(fullCommand);
+    }
+
+    private void parseTask(String fullCommand) throws DukeException{
+        switch (fullCommand.split(" ")[0]) {
+        case "todo":
+            String todoDescriptions = fullCommand.replace("todo", "");
+            if (todoDescriptions.trim().isEmpty()) {
+                throw new DukeException("Empty Todo Command!");
+            }
+            this.taskToAdd = new TodoTask(todoDescriptions.trim());
+            break;
+        case "deadline":
+            String[] deadlineDescriptions = fullCommand.replace("deadline", "").split("/by");
+            if (deadlineDescriptions[0].trim().isEmpty() || deadlineDescriptions[1].trim().isEmpty()) {
```

Comment:
Would it be better to declare the boolean as a variable on top? You can have a look at the Avoid Complicated Expressions section in this link. 
https://nus-cs2103-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html